File: passenger-checkin-service/src/main/java/com/airport/passenger_checkin_service/domain/entity/Passenger.java
package com.airport.passenger_checkin_service.domain.entity;

import com.airport.passenger_checkin_service.domain.enums.Gender;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.CompoundIndexes;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.time.LocalDate;

@Document(collection = "passengers")
@Data
@NoArgsConstructor
@AllArgsConstructor
@CompoundIndexes({
        @CompoundIndex(name = "unique_passport_idx",
                def = "{'passportNumber': 1, 'passportCountry': 1}",
                unique = true)
})
public class Passenger {
    @Id
    private ObjectId id;

    private String firstName;
    private String middleName;
    private String lastName;
    private Gender gender;
    private LocalDate dateOfBirth;
    private String email;
    private String phoneNumber;
    private String address;
    private String passportNumber;
    private String passportCountry;
    private LocalDate passportExpiryDate;
    private String nationality;

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}

File: airport-management-ui/src/types/passenger.ts
export enum Gender {
    MALE = 'MALE',
    FEMALE = 'FEMALE',
    OTHER = 'OTHER',
    UNSPECIFIED = 'UNSPECIFIED'
}

export interface Passenger {
    id: string;
    firstName: string;
    middleName?: string;
    lastName: string;
    gender: Gender;
    dateOfBirth: string;
    email?: string;
    phoneNumber?: string;
    address?: string;
    passportNumber: string;
    passportCountry: string;
    passportExpiryDate: string;
    nationality: string;
    createdAt: string;
    updatedAt: string;
}

export interface PassengerRequest {
    firstName: string;
    middleName?: string;
    lastName: string;
    gender: Gender;
    dateOfBirth: string;
    email?: string;
    phoneNumber?: string;
    address?: string;
    passportNumber: string;
    passportCountry: string;
    passportExpiryDate: string;
    nationality: string;
}

export interface PassengerSearchRequest {
    firstName?: string;
    middleName?: string;
    lastName?: string;
    gender?: Gender;
    address?: string;
    email?: string;
    phoneNumber?: string;
    passportNumber?: string;
    nationality?: string;
}

export interface PassengerResponse {
    id: string;
    firstName: string;
    middleName?: string;
    lastName: string;
    gender: Gender;
    dateOfBirth: string;
    email?: string;
    phoneNumber?: string;
    address?: string;
    passportNumber: string;
    passportCountry: string;
    passportExpiryDate: string;
    nationality: string;
    createdAt: string;
    updatedAt: string;
}